be1-go/channel/election/mod.go

```diff
@@ -819,12 +823,16 @@ func updateVote(msgID string, sender string, castVote messagedata.VoteCastVote,
 	return nil
 }
 
+// Creates the questions for the election channel or returns nil if they are not valid
 func getAllQuestionsForElectionChannel(questions []messagedata.ElectionSetupQuestion) map[string]*question {
 	qs := make(map[string]*question)
 	for _, q := range questions {
 		ballotOpts := make([]string, len(q.BallotOptions))
 		copy(ballotOpts, q.BallotOptions)
-
+		_, exists := qs[q.ID]
+		if exists {
+			return nil
```

Comment:
maybe log an error here?

Reply:
Thanks, I just added that in a new commit.

Reply:
Why not directly returning an error ? And checking the error value when the function is called ?

Reply:
The function is called when creating an election channel using a constructor. I tried implementing the function by having it return two values (the map and an error) but it seems that constructors in go don't allow you to initialize two fields in the same line (cf attached image).
<img width="578" alt="getAllQuestions" src="https://user-images.githubusercontent.com/53703829/222109068-9aefc7fb-a576-4cf3-8136-9379a09dce4a.png">

Reply:
You would need to initialize `questions` outside the constructor, the same way `registry` is done. You would get something like:

```
newChannel := &Channel{
...
}

questions, err := getAllQuestionsForElectionChannel()
if err != nil {
    return
}
newChannel.questions = questions
```